docs/proposals/mvp-dashboard-design.md

```diff
@@ -43,18 +43,82 @@ can query the relevant test builds.
+rate that test results enter the data storage. The server is to maintain a
+cache of frequently and recently accessed data. The server expects a json
+result from the data store.
+
+This data can then be received by the dashboard.
+
+### Dashboard
+
+The dashboard serves the purpose of allowing the user to make queries of the
+results stored in the data storage unit. The dashboard provides a simple user
+interface with a list of drop down menus for options such as: lab, config,
+test, run, and metric. The dashboard communicates with the server (both
+running on the same node) using an NGINX reverse proxy.
+
+The query is made via HTML requests in the form of `GET /api/labs`,
+`GET /api/labs/{lab id}`. The expected result is a json with the appropriate
+test data. This json is then used to create the graph of the specified metrics
+as they trend over the past 50 builds.
+
+Our choice of developing a REST API provides the benefit that any graph
+configured by a user can be sured via the URL link to that page.  This allows
+for the fast sharing of performance test results between team members and
+other teams.
 
 ## Rationale
 
-A discussion of alternate approaches and the trade offs, advantages, and
-disadvantages of the specified approach.
+1. Go vs. Python?
+1. Angular JS 2 vs Angular JS 1?
+1. Why build dashboard instead of using Grafana?
 
 ## Implementation
 
 A description of the steps in the implementation, who will do them, and when.
 
+### Writing the server API
+
+Upon approval, tasks will be identified and given to the performance team.
+
+### Building the dashboard
+
+Upon approval, tasks will be identified and given to the performance team.
+
 ## Open issues (if applicable)
 
-A discussion of issues relating to this proposal for which the author does not
-know the solution. This section may be omitted if there are none.
+1. What data storage service should be used to store the test results?
+  * The performance team has previously used [InfluxDB][1] to handle database
+    duties. InfluxDB is well-suited to the task due to its optimization for
+    time-series data. The perf team has also used it before so they are
+    already familiar with the InfluxDB libraries.
+  * Test results can be stored in files and orderedd in a series of
+    directories. This method would require minimal setup and can be easily
+    ported to another data storage system. An issue with this strategy is that
+    the data will have to all follow the same formatting standards, this adds
+    complexity and added time that the perf team will need to develop a format
+    that can accomodate all of the KPIs and their varying metrics.
+1. Should the server poll the data storage unit at a set rate or should the
+   data storage unit push updates whenever they occur?
+1. What tool will format and display the visualization of the data?
+  * [D3][2]
+  * [Angular Chart][3]
+  * [Grafana][4]
+  * Other tool?
```

Comment:
Again, our proposal should not leave this as an open issue. This needs to be addressed by it.